examples/data-objects/inventory-app/src/legacySharedTreeInventoryList.ts

```diff
@@ -0,0 +1,136 @@
+
+import { IInventoryListUntyped, IPart } from "./interfaces";
+
+const legacySharedTreeKey = "legacySharedTree";
+
+/**
+ * An inventory list implemented using LegacySharedTree.
+ */
+export class LegacySharedTreeInventoryList extends DataObject implements IInventoryListUntyped {
+	private _tree: LegacySharedTree | undefined;
+	private get tree() {
+		if (this._tree === undefined) {
+			throw new Error("Not initialized properly");
+		}
+		return this._tree;
+	}
+
+	protected async initializingFirstTime() {
+		const legacySharedTree = this.runtime.createChannel(
+			undefined,
+			LegacySharedTree.getFactory().type,
+		) as LegacySharedTree;
+
+		// Initialize the inventory with two parts at zero quantity
+		// REV: Is ChangeNode appropriate here? Or should I use some other type (BuildNode?)?
+		const inventoryNode: ChangeNode = {
+			identifier: legacySharedTree.generateNodeId(),
+			// REV: This is based on the Definition casting from Bubblebench, it doesn't really seem
+			// to matter what is used here though.  Is there a better way to omit the definition here?
+			definition: "array" as Definition,
+			traits: {
+				nut: [
+					{
+						identifier: legacySharedTree.generateNodeId(),
+						definition: "scalar" as Definition,
+						traits: {},
+						payload: 0,
+					},
+				],
+				bolt: [
+					{
+						identifier: legacySharedTree.generateNodeId(),
+						definition: "scalar" as Definition,
+						traits: {},
+						payload: 0,
+					},
+				],
+			},
+		};
+		legacySharedTree.applyEdit(
+			Change.insertTree(
+				inventoryNode,
+				StablePlace.atStartOf({
+					parent: legacySharedTree.currentView.root,
+					label: "parts" as TraitLabel,
+				}),
+			),
+		);
+
+		this.root.set(legacySharedTreeKey, legacySharedTree.handle);
+	}
+
+	protected async hasInitialized() {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		this._tree = await this.root
+			.get<IFluidHandle<LegacySharedTree>>(legacySharedTreeKey)!
+			.get();
+
+		// REV: I see the tsdoc for EditCommitted recommends using a Checkout instead but that seems
+		// like overkill (plus I don't see how it relates exactly).  Is there a better way to just observe
+		// that the scalars are changing and raise "inventoryChanged"?
+		this._tree.on(SharedTreeEvent.EditCommitted, () => {
+			this.emit("inventoryChanged");
+		});
+	}
+
+	// REV: Lots of casts and non-null assertions below - is there a more type-confident way to do this?
+	public getParts() {
+		const parts: IPart[] = [];
+		// REV: Seems strange that this.tree.currentView.rootNode is private.
+		const rootNode = this.tree.currentView.getViewNode(this.tree.currentView.root);
+		const partsNode = this.tree.currentView.getViewNode(
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			rootNode.traits.get("parts" as TraitLabel)![0],
+		);
+		// REV: This seems to iterate in reverse order?  Not sure why it swaps the entries as compared to how
+		// it was created.
```

Comment:
traits are un-ordered. For some technical reasons that I forget the details of, we sort them by label in a few places. That might be what's going on here.

Reply:
Gotcha - so if I wanted a stable display order, is there a best practice?  I assume I'll need to add a node in the tree somewhere with some sortable characteristic.

Maybe I shouldn't be using part names for the traits, but instead have `inventory->traits->parts[]->part->traits->name,quantity,sortableThing`?

Reply:
this aligns with your thought. FWIW, I generally find YAML-style to be convenient expression of the domain model by giving a sample tree:

```
Inventory
  parts:
    - Part
        type: "nut",
        quantity: 5
    - Part
        type: "bolt"
        quantity: 1
```

Reply:
Updated to have this structure - this incidentally caused the order to flip :-P

Re: using sequence APIs to control node order - do you have a pointer to an example of that?  Would it be something like a series of `Change.insertTree()` calls per part when building the tree rather than putting them all in with a single `BuildNode`?

Reply:
I think some of [this section of the readme](https://github.com/microsoft/FluidFramework/tree/main/experimental/dds/tree#editing) is a little out of date (I think the imperative API might've been removed?), but the section on StablePlace still looks good to me and is what you want: two concurrent edits which try to insert content "before the end of a trait" will end up with later-write-is-later-in-the-sequence, whereas the same edits which try to insert content "after the (current) last node in that trait" will end up with later-write-is-earlier-in-the-sequence. 

That said, it doesn't look like you're modifying the sequence past initial insert, so it shouldn't really matter what you do here. What order are you getting in the current version of the code? is it not `[nut, bolt]`?

With the domain model suggested above, in particular I wouldn't expect this to be necessary:

> 		// REV: This is not a stable/controlled ordering - if we want to control the order, we might add
>		// another node under the partNode that specifies some payload to order on.

that should just happen for free.

Reply:
Sounds good, I'll try that out.  I recognize that this is legacy shared tree so maybe not getting investment at the moment, but it is surprising to me that the provided order isn't preserved (basically implicitly doing this approach).

I am getting `[nut, bolt]` currently.  My best guess is that the natural order has something to do with the specific nodeIds that are getting assigned, which changed when the tree structure changed

Reply:
I'd expect `[nut, bolt]` in your current code because:

```

		const inventoryNode: BuildNode = {
			definition: "inventory",
			traits: {
				parts: [
					{
						definition: "part",
						traits: {
							name: {
								definition: "name",
								payload: "nut",
							},
							quantity: {
								definition: "quantity",
								payload: 0,
							},
						},
					},
					{
						definition: "part",
						traits: {
							name: {
								definition: "name",
								payload: "bolt",
							},
							quantity: {
								definition: "quantity",
								payload: 0,
							},
						},
					},
				],
			},
		};
		legacySharedTree.applyEdit(
			Change.insertTree(
				inventoryNode,
				StablePlace.atStartOf({
					parent: legacySharedTree.currentView.root,
					label: "inventory" as TraitLabel,
				}),
			),
		);
```

and the contents of the trait "parts" are ordered (which is consistent with the fact they're in an array)

Reply:
Gotcha - so would it be fair to say "The enumeration order of traits on a node is stable but arbitrary, but the order of trait contents (if a sequence) is as-specified"?

Reply:
Yep.